cmd/net.go

```diff
@@ -65,12 +67,42 @@ func mustGetLocalIP4() (ipList set.StringSet) {
 
 // getHostIP4 returns IPv4 address of given host.
 func getHostIP4(host string) (ipList set.StringSet, err error) {
-	ipList = set.NewStringSet()
-	ips, err := net.LookupIP(host)
-	if err != nil {
-		return ipList, err
+	var ips []net.IP
+
+	if ips, err = net.LookupIP(host); err != nil {
+		// return err if not Docker or Kubernetes
+		// We use IsDocker() method to check for Docker Swarm environment
+		// as there is no reliable way to clearly identify Swarm from
+		// Docker environment.
+		if !IsDocker() && !IsKubernetes() {
+			return ipList, err
+		}
+
+		// channel to indicate completion of host resolution
+		doneCh := make(chan struct{})
+		// Indicate retry routine to exit cleanly, upon this function return.
+		defer close(doneCh)
+		// Mark the starting time
+		startTime := time.Now()
+		// wait for hosts to resolve in exponentialbackoff manner
+		for _ = range newRetryTimerSimple(doneCh) {
+			// Retry infinitely on Kubernetes and Docker swarm.
+			// This is needed as the remote hosts are sometime
+			// not available immediately.
+			if ips, err = net.LookupIP(host); err == nil {
+				break
+			}
+			// log error only if more than 1s elapsed
+			if time.Since(startTime) > time.Second {
+				// log the message to console about the host not being
+				// resolveable.
+				errorIf(err, "Unable to resolve host %s (%s)", host,
+					humanize.RelTime(startTime, time.Now(), "elapsed", ""))
```

Comment:
Instead of using `time.Now()` twice (`time.Since()` is short form of `time.Now().Sub(t)`), you could use below technique
```go
startTime := time.Now()
timeZero := time.Time{}
....
....
older := time.Since(startTime)
if older > time.Second {
    errorIf(err, "Unable to resolve host %s (%s)", host, humanize.RelTime(timeZero, timeZero.Add(older), "elapsed", ""))
}
```